pex/interpreter.py

```diff
@@ -183,10 +184,23 @@ def get(cls, binary=None):
         # vendored attrs distribution so that its `cache_hash=True` feature can work (see the
         # bottom of pex/third_party/__init__.py where the vendor importer is installed). We ignore
         # such adjoined `sys.path` entries to discover the true base interpreter `sys.path`.
-        pythonpath = frozenset(
-            os.environ.get("PYTHONPATH", "").split(os.pathsep) + list(third_party.exposed())
+        pythonpath = os.environ.get("PYTHONPATH")
+        internal_entries = frozenset(
+            (pythonpath.split(os.pathsep) if pythonpath else []) + list(third_party.exposed())
         )
-        sys_path = OrderedSet(item for item in sys.path if item and item not in pythonpath)
+
+        def is_internal_entry(entry):
+            # type: (str) -> bool
+            if entry in internal_entries:
+                return True
+            if not os.path.isabs(entry):
```

Comment:
Might `internal_entries` also have relative paths in it (e.g. `PYTHONPATH="./foo/bar"`)? Does that need handling?

Reply:
It cannot, no.